picard/webservice/__init__.py

```diff
@@ -65,6 +66,174 @@ def __init__(self, response_type):
+        and disables caching.
+    :param access_token: OAuth token.
+    :param queryargs: `dict` of query arguments.
+    :param retries: Current retry attempt number.
+    :param priority: Indicates that this is a high priority request.  (See
+        `WebService._run_next_task`.)
+    :param important: Indicates that this is an important request.  (Ditto.)
+    """
+
+    def __init__(self, method, host, port, path, handler, parse_response_type=None, data=None,
+                 mblogin=False, cacheloadcontrol=None, refresh=None,
+                 queryargs=None, priority=False, important=False):
+        url = build_qurl(host, port, path=path, queryargs=queryargs)
+        super().__init__(url)
+
+        # These two are codependent (see _update_authorization_header) and must
+        # be initialized explicitly.
+        self._access_token = None
+        self._mblogin = None
+
+        self._retries = 0
+
+        self.method = method
+        self.host = host
+        self.port = port
+        self.path = path
+        self.handler = handler
+        self.parse_response_type = parse_response_type
+        self.response_parser = None
+        self.response_mimetype = None
+        self.data = data
+        self.mblogin = mblogin
+        self.cacheloadcontrol = cacheloadcontrol
+        self.refresh = refresh
+        self.queryargs = queryargs
+        self.priority = priority
+        self.important = important
+
+        self.access_token = None
+        self._init_headers()
+
+    def _init_headers(self):
+        self.setHeader(QtNetwork.QNetworkRequest.UserAgentHeader, USER_AGENT_STRING)
+
+        if self.mblogin:
+            self.setPriority(QtNetwork.QNetworkRequest.HighPriority)
+            self.setAttribute(QtNetwork.QNetworkRequest.CacheLoadControlAttribute,
+                                 QtNetwork.QNetworkRequest.AlwaysNetwork)
+        elif self.cacheloadcontrol is not None:
+            self.setAttribute(QtNetwork.QNetworkRequest.CacheLoadControlAttribute,
+                                 self.cacheloadcontrol)
+
+        if self.parse_response_type:
+            try:
+                self.response_mimetype = WebService.get_response_mimetype(self.parse_response_type)
+                self.response_parser = WebService.get_response_parser(self.parse_response_type)
+            except UnknownResponseParserError as e:
+                log.error(e.args[0])
+            else:
+                self.setRawHeader(b"Accept", self.response_mimetype.encode('utf-8'))
+
+        if self.data:
+                self.setHeader(QtNetwork.QNetworkRequest.ContentTypeHeader, "application/x-www-form-urlencoded")
+
+    def _update_authorization_header(self):
+        if self.mblogin and self.access_token:
+            self.setRawHeader(b"Authorization", ("Bearer %s" % string_(self.access_token)).encode('utf-8'))
+        else:
+            self.setRawHeader(b"Authorization", b"")
```

Comment:
https://code.woboq.org/qt5/qtbase/src/network/access/qnetworkrequest.cpp.html#1034 suggests a null value would remove the header.
@samj1912 : can you verify the header is actually removed with `b""`